src/lib/axios.ts

```diff
@@ -1,8 +1,121 @@
-import axios from "axios"
+import {
+	AnnouncementResponse,
+	EventsResponse,
+	NewsResponse,
+	SingleAnnouncementResponse,
+	SingleEventResponse,
+	SingleNewsResponse,
+	SubstitutionsResponse,
+} from "@/types/axios"
+import axios, { AxiosError } from "axios"
+import { format } from "date-fns"
 
 export const api = axios.create({
-	baseURL: "https://jsonplaceholder.typicode.com",
+	baseURL: "http://api.elektronik.zgora.pl/api",
 	headers: {
 		"Content-Type": "application/json",
 	},
 })
+
+function handleError(error: AxiosError) {
```

Comment:
trochę słabe są te try catche wszędzie, lepiej będzie coś w tym stylu zrobić:
```ts
const getUser = async (id: number) => {}

function catchError<T>(promise: Promise<T>): Promise<[undefined, T] | [Error]> {
    return promise
        .then((data) => [undefined, data] as [undefined, T])
        .catch((error) => [error])
}

const [data, error] = await catchError(getUser(1))
```

Reply:
No właśnie cos podobnego mamy w tanstacku co nie

```tsx
function Todos() {
  const { isPending, isError, data, error } = useQuery({
    queryKey: ['todos'],
    queryFn: fetchTodoList,
  })

  if (isPending) {
    return <span>Loading...</span>
  }

  if (isError) {
    return <span>Error: {error.message}</span>
  }

  // We can assume by this point that `isSuccess === true`
  return (
    <ul>
      {data.map((todo) => (
        <li key={todo.id}>{todo.title}</li>
      ))}
    </ul>
  )
}
```
Więc data + state management jest już zapewniony, a to co napisałeś to chyba ma na celu właśnie coś podobnego.